### PR TITLE
Add optional SSL/TLS support to server and metrics net listeners

### PR DESCRIPTION
**Pull request by timvaillancourt**
As many/most Redis drivers support TLS connections, I've added support to listen with TLS for both the Titan server and metrics listeners.

Notes:
1. Disabled by default.
1. Enabled by defining a TLS certificate and key file.
    - Enabled for titan server and metrics listeners separately.
1. Simple unit-tests test functionality.

EDIT: I plan to send a 2nd PR to enable TLS between Titan and the TiKv store.

### CODE REVIEW

File: server/testdata/server.crt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+-----BEGIN CERTIFICATE-----
```
Comment (arthurkiller) [inline server/testdata/server.crt]: BTW @timvaillancourt 
You can generate the test file when running the unit test.

```
fun genCRTTest(t *testing.T) {
    // do something just like echo "data" >> server.crt
}

func cleanCRTtest(t *testing.T) {
    // do something like rm
}
```

File: bin/titan/main.go [hunk 2]
```diff
@@ -74,16 +75,38 @@ func main() {
 		ListZipThreshold: config.Server.ListZipThreshold,
 	})
 
+	// titan server tls options
+	var servTLSOpts continuous.ServerOption
+	if config.Server.TLSCertFile != "" && config.Server.TLSKeyFile != "" {
+		tlsConfig, err := server.TLSConfig(config.Server.TLSCertFile, config.Server.TLSKeyFile)
+		if err != nil {
+			fmt.Printf("failed to load server TLS config: %s\n", err)
```
Comment (YIDWang) [inline bin/titan/main.go]: Zap is recommended for logging.
Fatal contains both fmt and exit operations.
The global logger prints to the normal output by default until there is no redirection.

File: bin/titan/main.go [hunk 3]
```diff
@@ -74,16 +75,38 @@ func main() {
...
+		}
+		servTLSOpts = continuous.TLSConfig(tlsConfig)
+	}
+
+	// status server tls options
+	var statusTLSOpts continuous.ServerOption
+	if config.Status.TLSCertFile != "" && config.Status.TLSKeyFile != "" {
+		tlsConfig, err := server.TLSConfig(config.Status.TLSCertFile, config.Status.TLSKeyFile)
+		if err != nil {
+			fmt.Printf("failed to load status server TLS config: %s\n", err)
```
Comment (YIDWang) [inline bin/titan/main.go]: Same as above.

**Comment by timvaillancourt** [comment]
Please hold on merging. TiDB uses the word "ssl" instead of "tls" so I think I should match that in the titan config, eg: "ssl-cert-file" instead of "tls-cert-file". Thoughts?

https://godoc.org/github.com/pingcap/tidb/config#Security

**Comment by arthurkiller** [comment]
> Please hold on merging. TiDB uses the word "ssl" instead of "tls" so I think I should match that in the titan config, eg: "ssl-cert-file" instead of "tls-cert-file". Thoughts?
> 
> https://godoc.org/github.com/pingcap/tidb/config#Security

NMV, but whether `tls` or `ssl` is almost the same thing, `tls 1.0` is an alias to `ssl 3.1`.

**Comment by arthurkiller** [comment]
BTW, you can add a `WIP:` before the PR title, and remove it after you have your work done.

**Comment by timvaillancourt** [comment]
Thanks, that rename is complete. This now matches TiKV naming.

**Pull request merged**